doc/guides/timers-in-node.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Timers in node.js
+layout: docs.hbs
+---
+
+# Timers in node.js and beyond
+
+Timers are a collection of global functions in node.js that allow you to 
+execute code after a set period of time. To fully understand when timer 
+functions will be executed, it's a good idea to read up on the the node 
+[Event Loop](https://nodesource.com/blog/understanding-the-nodejs-event-loop/).
+
+## Code Time Machine
+
+The node API provides several ways to schedule your code to execute at 
+some point after the present moment.
+
+### "When I say so" Execution ~ *setTimeout()*
+
+Using `setTimeout()`, you can schedule code to execute after a designated 
+amount of milliseconds. You may already be familiar with this function, as 
+it is a part of V8 and thus part of the browser JavaScript API.
+
+`setTimeout()` accepts the code to execute as its first argument and the 
+millisecond delay defined as a number literal as the second argument. Here 
+is an example of that:
+
+```js
+function myFunc () {
+  console.log('funky');
+}
+
+setTimeout(myFunc, 1500);
+```
+
+The above function `myFunc()` will execute after approximately 1500 
+milliseconds (or 1.5 seconds) due to the call of `setTimeout()`.
+
+The timeout interval that is set cannot be relied upon to execute after 
+that *exact* number of milliseconds. This is because executing code that 
+blocks or holds onto the event loop will push the execution of your timeout 
+back. You *can* guarantee that your timeout will not execute *sooner* than 
+the declared timeout.
+
+### "Right after this" Execution ~ *setImmediate()*
+
+`setImmediate()` allows you to execute code at the beginning of the next 
+event loop cycle. This code will execute *before* any timers or IO operations. 
+I like to think of this code execution as happening "right after this", meaning 
+any code following the `setImmediate()` function call will execute before the 
+`setImmediate()` function argument. Here's an example:
+
+```js
+let order = 'before immediate\n';
+
+setImmediate(() => {
+  order += 'executing immediate\n';
+});
+
+order += 'after immediate\n';
+
+console.log(order);
+```
+
+The above function passed to `setImmediate()` will execute after all runnable 
+code has executed, and the console output will be:
+
+```shell
+before immediate
+after immediate
+executing immediate
+```
+
+Note: `process.nextTick()` is very similar to `setImmediate()`. The two major 
```

Comment:
Not really. `setImmediate()` is what most people will think `nextTick()` does. (Indeed, what `nextTick()` **used** to do.)

The primary difference is that immediates allows IO to run before they are called, whereas nextTick does not. (nextTick is also not safe to call recursively in a potentially unbounded (infinite) way.)